soc/riscv/riscv-privilege/nuclei/common/nuclei_csr.h

```diff
@@ -0,0 +1,252 @@
+
+#define MMISC_CTL_NMI_CAUSE_FFF     (1 << 9)
+#define MMISC_CTL_MISALIGN          (1 << 6)
+#define MMISC_CTL_BPU               (1 << 3)
+
+#define MCACHE_CTL_IC_EN            (1 << 0)
+#define MCACHE_CTL_IC_SCPD_MOD      (1 << 1)
+#define MCACHE_CTL_IC_ECC_EN        (1 << 2)
+#define MCACHE_CTL_IC_ECC_EXCP_EN   (1 << 3)
+#define MCACHE_CTL_IC_RWTECC        (1 << 4)
+#define MCACHE_CTL_IC_RWDECC        (1 << 5)
+#define MCACHE_CTL_DC_EN            (1 << 16)
+#define MCACHE_CTL_DC_ECC_EN        (1 << 17)
+#define MCACHE_CTL_DC_ECC_EXCP_EN   (1 << 18)
+#define MCACHE_CTL_DC_RWTECC        (1 << 19)
+#define MCACHE_CTL_DC_RWDECC        (1 << 20)
+
+#define MTVT2_MTVT2EN               (1 << 0)
+#define MTVT2_COMMON_CODE_ENTRY     (((1ULL << ((__riscv_xlen) - 2)) - 1) << 2)
+
+#define MCFG_INFO_TEE               (1 << 0)
+#define MCFG_INFO_ECC               (1 << 1)
+#define MCFG_INFO_CLIC              (1 << 2)
+#define MCFG_INFO_PLIC              (1 << 3)
+#define MCFG_INFO_FIO               (1 << 4)
+#define MCFG_INFO_PPI               (1 << 5)
+#define MCFG_INFO_NICE              (1 << 6)
+#define MCFG_INFO_ILM               (1 << 7)
+#define MCFG_INFO_DLM               (1 << 8)
+#define MCFG_INFO_ICACHE            (1 << 9)
+#define MCFG_INFO_DCACHE            (1 << 10)
+
+#define MICFG_IC_SET                (0xF << 0)
+#define MICFG_IC_WAY                (0x7 << 4)
+#define MICFG_IC_LSIZE              (0x7 << 7)
+#define MICFG_IC_ECC                (0x1 << 10)
+#define MICFG_ILM_SIZE              (0x1F << 16)
+#define MICFG_ILM_XONLY             (0x1 << 21)
+#define MICFG_ILM_ECC               (0x1 << 22)
+
+#define MDCFG_DC_SET                (0xF << 0)
+#define MDCFG_DC_WAY                (0x7 << 4)
+#define MDCFG_DC_LSIZE              (0x7 << 7)
+#define MDCFG_DC_ECC                (0x1 << 10)
+#define MDCFG_DLM_SIZE              (0x1F << 16)
+#define MDCFG_DLM_ECC               (0x1 << 21)
+
+#define MPPICFG_INFO_PPI_SIZE       (0x1F << 1)
+#define MPPICFG_INFO_PPI_BPA        (((1ULL << ((__riscv_xlen) - 10)) - 1) << 10)
+
+#define MFIOCFG_INFO_FIO_SIZE       (0x1F << 1)
+#define MFIOCFG_INFO_FIO_BPA        (((1ULL << ((__riscv_xlen) - 10)) - 1) << 10)
+
+#define MECC_LOCK_ECC_LOCK          (0x1)
+
+#define MECC_CODE_CODE              (0x1FF)
+#define MECC_CODE_RAMID             (0x1F << 16)
+#define MECC_CODE_SRAMID            (0x1F << 24)
+
+#define CCM_SUEN_SUEN               (0x1 << 0)
+#define CCM_DATA_DATA               (0x7 << 0)
+#define CCM_COMMAND_COMMAND         (0x1F << 0)
+
+#define FRM_RNDMODE_RNE     0x0
+#define FRM_RNDMODE_RTZ     0x1
+#define FRM_RNDMODE_RDN     0x2
+#define FRM_RNDMODE_RUP     0x3
+#define FRM_RNDMODE_RMM     0x4
+#define FRM_RNDMODE_DYN     0x7
+
+#define FFLAGS_AE_NX        (1 << 0)
+#define FFLAGS_AE_UF        (1 << 1)
+#define FFLAGS_AE_OF        (1 << 2)
+#define FFLAGS_AE_DZ        (1 << 3)
+#define FFLAGS_AE_NV        (1 << 4)
+
+#ifdef __riscv
+
+#ifdef __riscv64
```

Comment:
is riscv64 meaningful here?

A description of what this file adds would be good.

Reply:
This file is cherry-picking from nuclei SDK's definitions to define Nuclei specific CSR and related defines.
Remove the __risc64 condition because we are not using 64bit Nuclei SoC.
And I wrote a brief description of this file at the beginning of this file.